### PR TITLE
Enable positive group indices for extractAllRecord on JNI

### PR DESCRIPTION
**Pull request by anthony-chang**
Closes https://github.com/rapidsai/cudf/issues/11033

Allow `extractAllRecord` to take an `idx` > 0 and exposes `cudf::strings::extract_all_record`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 1]
```diff
@@ -3246,13 +3246,12 @@ public final Table extractRe(String pattern) throws CudfException {
    * @link https://docs.rapids.ai/api/libcudf/nightly/md_regex.html
    
    * @param pattern The regex pattern
-   * @param idx The regex group index (only 0 is supported currently). 
+   * @param idx The regex group index.
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:3249]: Just small nit to make this consistent with other lines.
```suggestion
   * @param idx The regex group index
```

**Review by ttnghia — APPROVED** [review]

**Review by mythrocks — APPROVED** [review]
LGTM.

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 2]
```diff
@@ -4111,6 +4111,7 @@ void testExtractRe() {
 
   @Test
   void testExtractAllRecord() {
+    String pattern = "([ab])(\\d)";
```
Comment (abellina) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:4114]: for my edification, so group index 0 in this case is `([ab])`?

Comment (anthony-chang) [reply java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:4114]: 0 is a special case in Spark that matches the entire regular expression (all the groups). The group indices are 1-indexed.

**Review by abellina — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

**Comment by revans2** [comment]
@gpucibot merge

**Pull request merged**